comb_spec_searcher/proof_tree.py

```diff
@@ -231,14 +231,9 @@ def __eq__(self, other):
 
     def get_function(self, min_poly=False):
         if min_poly:
-            if (self.sympy_function is None or
-                    isinstance(self.sympy_function, sympy.Function)):
-                self.sympy_function = sympy.var("F_" + str(self.label))
+            return sympy.var("F_" + str(self.label))
         else:
-            if (self.sympy_function is None or
-                    isinstance(self.sympy_function, sympy.Symbol)):
-                self.sympy_function = sympy.Function(
-                                        "F_" + str(self.label))(sympy.abc.x)
+            return sympy.Function("F_" + str(self.label))(sympy.abc.x)
         return self.sympy_function
```

Comment:
This line should be removed, then it is ready!

Reply:
Good I changed it